var/spack/repos/builtin/packages/aomp/package.py

```diff
@@ -119,20 +119,17 @@ class Aomp(Package):
     depends_on('libffi', type=('build', 'link'))
 
     for ver in ['3.5.0', '3.7.0', '3.8.0', '3.9.0', '3.10.0']:
-        depends_on('hsakmt-roct@' + ver, type=('build', 'run'), when='@' + ver)
+        depends_on('hsakmt-roct@' + ver, when='@' + ver)
         depends_on('comgr@' + ver, type='build', when='@' + ver)
-        depends_on('hsa-rocr-dev@' + ver, type=('build', 'run'),
-                   when='@' + ver)
-        depends_on('rocm-device-libs@' + ver, type=('build', 'run'),
-                   when='@' + ver)
+        depends_on('hsa-rocr-dev@' + ver, when='@' + ver)
+        depends_on('rocm-device-libs@' + ver, when='@' + ver)
 
         if ver != '3.5.0':
-            depends_on('hip@' + ver, type=('build', 'run'), when='@' + ver)
-            depends_on('hip-rocclr@' + ver, type='build', when='@' + ver)
+            depends_on('hip@' + ver, when='@' + ver)
+            depends_on('hip-rocclr@' + ver, when='@' + ver)
 
         if ver == '3.9.0' or ver == '3.10.0':
-            depends_on('rocm-gdb@' + ver, type=('build', 'run'),
-                       when='@' + ver)
+            depends_on('rocm-gdb@' + ver, when='@' + ver)
```

Comment:
```spack spec aomp``` gives an error for me:
==> Error: An unsatisfiable variant constraint has been detected for spec
hip requires llvm-amdgpu variant +rocm-device-libs, but spec asked for ~rocm-device-libs

For the aomp package, we can get rid of dependencies on hip, hip-rocclr and rocm-gdb. Lines(127-132)

Reply:
You may need to use the new concretizer (run spack config edit config and add `concretizer: clingo` to the `config:` section), since the old concretizer is not very smart.